src/documentation/io/authoring-overview.md

```diff
@@ -10,35 +10,80 @@ permalink: /documentation/io/authoring-overview/
 
-> Note: This guide is still in progress. There is an open issue to finish the guide: [BEAM-1025](https://issues.apache.org/jira/browse/BEAM-1025).
 
 * TOC
 {:toc}
 
 ## Introduction
-TODO
+This guide covers how to implement I/O transforms in the Beam model. Beam pipelines use these read and write transforms to import data for processing, and write data to a store.
+
+Reading and writing data in Beam is a parallel task, and using `ParDo`s, `GroupByKey`s, etc... is usually sufficient. Rarely, you will need the more specialized `Source` and `Sink` classes for specific features. There are changes coming soon (`SplittableDoFn`, [BEAM-65](https://issues.apache.org/jira/browse/BEAM-65)) that will make `Source` unnecessary.
+
+As you work on your I/O Transform, be aware that the Beam community is excited to help those building new I/O Transforms and that there are many examples and helper classes.
 
-## Example I/O Transforms
-TODO
 
 ## Suggested steps for implementers
-TODO
+1. Check out this guide and come up with your design. If you'd like, you can email the [Beam dev mailing list]({{ site.baseurl }}/get-started/support) with any questions you might have. It's good to check there to see if anyone else is working on the same I/O Transform.
+2. If you are planning to contribute your I/O transform to the Beam community, you'll be going through the normal Beam contribution life cycle - see the [Apache Beam Contribution Guide]({{ site.baseurl }}/contribute/contribution-guide/) for more details.
+3. As you're working on your IO transform, see the [PTransform Style Guide]({{ site.baseurl }}/contribute/ptransform-style-guide/) for specific information about writing I/O Transforms.
+
 
 ## Read transforms
-TODO
+Read transforms take data from outside of the Beam pipeline and produce PCollections of data.
+
+For data stores or files types where the data can be read in parallel, you can think of the process as a mini-pipeline. This often consists of two steps:
+1. Splitting the data into parts to be read in parallel
+2. Reading from each of those parts
+
+Each of those steps will be a `ParDo`, with a `GroupByKey` in between. The `GroupByKey` is an implementation detail, but for most runners it allows the runner to use different numbers of workers for:
+* Determining how to split up the data to be read into chunks - this will likely occur on very few workers
+* Reading - will likely benefit from more workers
+
+The `GroupByKey` will also allow Dynamic Work Rebalancing to occur (on supported runners).
+
+Here are some examples of read transform implementations that use the "reading as a mini-pipeline" model when data can be read in parallel:
+* **Reading from a file glob** - For example reading all files in "~/data/**"
+  * Get File Paths `ParDo`: As input, take in a file glob. Emit a PCollection of strings, each of which is a file path.
+  * Reading `ParDo`: Given the PCollection of file paths, read each one, emitting a PCollection of records.
```

Comment:
You emit individual records, not collections - applying the ParDo produces a PCollection.

Perhaps: Given the PCollection of file paths, apply to it a ParDo that reads records in each file, producing a PCollection of records.

Rephrase previous bullet similarly, and likewise below